src/components/dialogs/network-modifications/generator/creation/generator-creation-dialog.tsx

```diff
@@ -275,8 +355,8 @@ const GeneratorCreationDialog = ({
                 voltageRegulationOn: generator[VOLTAGE_REGULATION],
                 targetV: generator[VOLTAGE_SET_POINT],
                 qPercent: generator[Q_PERCENT],
-                voltageLevelId: generator[CONNECTIVITY]?.[VOLTAGE_LEVEL]?.[ID],
-                busOrBusbarSectionId: generator[CONNECTIVITY]?.[BUS_OR_BUSBAR_SECTION]?.[ID],
+                voltageLevelId: generator.connectivity.voltageLevel.id,
+                busOrBusbarSectionId: generator.connectivity.busOrBusbarSection.id,
                 isUpdate: !!editData,
                 modificationUuid: editData?.uuid ?? null,
                 plannedActivePowerSetPoint: generator[PLANNED_ACTIVE_POWER_SET_POINT],
```

Comment:
in submit, should post an object with type conform to one in the back, do not transform again in the api call which can lead to error..